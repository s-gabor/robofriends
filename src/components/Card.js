import React from 'react';

const Card = ({ name, email }) => {
    // if(name === 'Clementine Bauch') {
    //     throw new Error();
    // }
    return (
        <div className='bg-light m-3 shadow rounded grow'>
            <img
                className='rounded-top'
                src={`https://robohash.org/${name}.png?size=220x220&bgset=bg1`}
                alt='robots'
            />
            <div className='text-center'>
                <h5>{name}</h5>
                <p>{email}</p>
            </div>
        </div>
    )
}

export default Card;
