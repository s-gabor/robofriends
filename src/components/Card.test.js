import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

// shuld be inside a setupTests.js file inside containers
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

it('expect to render Card component', () => {
    const wraper = shallow(<Card />);
    expect(wraper).toMatchSnapshot();
});
