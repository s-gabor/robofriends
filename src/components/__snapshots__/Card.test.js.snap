// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card component correctly 1`] = `
<div
  className="bg-light m-3 shadow rounded grow"
>
  <img
    alt="robots"
    className="rounded-top"
    src="https://robohash.org/undefined.png?size=220x220&bgset=bg1"
  />
  <div
    className="text-center"
  >
    <h5 />
    <p />
  </div>
</div>
`;
